ee/hogai/graph/taxonomy/toolkit.py

```diff
@@ -284,45 +284,85 @@ def _get_custom_tools(self) -> list:
         """Get custom tools. Override in subclasses to add custom tools."""
         raise NotImplementedError("_get_custom_tools must be implemented in subclasses")
 
-    def retrieve_entity_properties(self, entity: str, max_properties: int = 500) -> str:
+    def retrieve_entity_properties(self, entity: str | list[str], max_properties: int = 500) -> dict[str, str]:
```

Comment:
This function is just a wrapper for `_bulk_get_entity_properties` with a transformation from str to list[str], seems like we can move that transformation to the bulk function and remove this one.